opennms-config/src/main/java/org/opennms/netmgt/config/PollerConfigManager.java

```diff
@@ -133,7 +133,9 @@ protected void setUpInternalData() {
      * @throws java.io.IOException if any.
      */
     @Override
-    public abstract void update() throws IOException;
+    public void update() throws IOException {
+        setUpInternalData();
```

Comment:
We may need to review the locking here. Looks like there may be cases when callers can get an invalid state (`createUrlIpMap` doesn't lock,)

Reply:
Sure. Since this is only updating internal state and not changing anything in config. added read lock for `update`.